corehq/apps/case_search/utils.py

```diff
@@ -195,7 +178,9 @@ def __init__(self, domain):
         self.domain = domain
         self.profiler = CaseSearchProfiler()
 
-    def get_base_queryset(self):
+    def get_base_queryset(self, slug=None):
+        # slug is only informational, used for profiling
+        CaseSearchES = self.profiler.get_case_search_class(slug)
```

Comment:
I'm not certain whether this is the best way to override the class, but I do like the idea of using an auto-logging subclass rather than needing to use the profiler to call `run`.  Open to feedback if you can think of a cleaner approach

Reply:
@gherceg you might have opinions here - I know you also looked at various options for logging ES queries in case search